src/define_diffusion.jl

```diff
@@ -86,14 +86,14 @@ function HorizontalDiffusion(   P::Parameters,          # Parameter struct
         end
     end
 
-    if P.model == :barotropic || P.model == :shallowwater               # orographic correction not needed
+    if P.model <: BarotropicModel || P.model <: ShallowWaterModel       # orographic correction not needed
```

Comment:
I'd assume that use of the model types here, would always require the added abstract types. This file needs to be called before the file that defines the models, as it needs `HorizontalDiffusion`